File: engine/model/src/main/scala/tools/packet/npctalk/GetNPCTalkNum.scala
package tools.packet.npctalk

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetNPCTalkNum(private var _npcId: Int, private var _talk: String, private var _def: Int, private var _min: Int, private var _max: Int) extends PacketInput {
  def npcId: Int = _npcId

  def talk: String = _talk

  def theDef: Int = _def

  def min: Int = _min

  def max: Int = _max

  override def opcode(): SendOpcode = SendOpcode.NPC_TALK
}

File: engine/engine-model/src/main/java/tools/packet/npctalk/GetNPCTalkNum.java
package tools.packet.npctalk;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetNPCTalkNum(Integer npcId, String talk, Integer theDef, Integer min,
                            Integer max) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NPC_TALK;
   }
}